examples/hello-windows/main.rs

```diff
@@ -0,0 +1,220 @@
+use std::collections::HashMap;
+use winit::{
+    event::{Event, WindowEvent},
+    event_loop::{ControlFlow, EventLoop},
+    window::{Window, WindowId},
+};
+
+struct ViewportDesc<S = ()> {
+    window: Window,
+    background: wgpu::Color,
+    surface: S,
+}
+
+struct Viewport {
+    desc: ViewportDesc<wgpu::Surface>,
+    sc_desc: wgpu::SwapChainDescriptor,
+    swap_chain: wgpu::SwapChain,
+}
+
+impl ViewportDesc {
+    fn new(window: Window, background: wgpu::Color) -> Self {
+        Self {
+            window,
+            background,
+            surface: (),
+        }
+    }
+    fn create_surface(self, instance: &wgpu::Instance) -> ViewportDesc<wgpu::Surface> {
+        let Self {
+            window, background, ..
+        } = self;
+        let surface = unsafe { instance.create_surface(&window) };
+        ViewportDesc {
+            window,
+            surface,
+            background,
+        }
+    }
+}
+
+impl ViewportDesc<wgpu::Surface> {
+    fn build(self, device: &wgpu::Device, swapchain_format: wgpu::TextureFormat) -> Viewport {
+        let size = self.window.inner_size();
+
+        let sc_desc = wgpu::SwapChainDescriptor {
+            usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+            format: swapchain_format,
+            width: size.width,
+            height: size.height,
+            present_mode: wgpu::PresentMode::Fifo,
+        };
+
+        let swap_chain = device.create_swap_chain(&self.surface, &sc_desc);
+
+        Viewport {
+            desc: self,
+            sc_desc,
+            swap_chain,
+        }
+    }
+}
+
+impl Viewport {
+    fn resize(&mut self, device: &wgpu::Device, size: winit::dpi::PhysicalSize<u32>) {
+        self.sc_desc.width = size.width;
+        self.sc_desc.height = size.height;
+        self.swap_chain = device.create_swap_chain(&self.desc.surface, &self.sc_desc);
+    }
+    fn get_current_frame(&mut self) -> wgpu::SwapChainTexture {
+        self.swap_chain
+            .get_current_frame()
+            .expect("Failed to acquire next swap chain texture")
+            .output
+    }
+}
+
+async fn run(
+    event_loop: EventLoop<()>,
+    viewports: Vec<ViewportDesc>,
```

Comment:
I think it would be easier to not make `ViewportDesc` generic, and just have `Vec<(Window, wgpu::Color)>` here

Reply:
Ok, done. It's fine to simplify example, but in real-life situation there would be much more parameters other than just `background`.